superset-frontend/src/views/components/Menu.tsx

```diff
@@ -284,6 +298,7 @@ export function Menu({
             className="main-nav"
           >
             {menu.map(item => {
+              if (item.name === 'Upload Data') return null;
```

Comment:
wdyt is better? Having a null item in the array or filtering the value out before mapping? I'm not sure what an empty element does in the array here for the menu, but it seems that maybe just removing it would be cleaner.

Reply:
On second thought, after reading further down, same sort of suggestion applies here. If we don't want to render this menu because it has a submenu, can we use the existence of a submenu as a reason not to render it rather than the name itself, in case we want to change the name later or add more submenus?